### PR TITLE
Add table alicloud_cs_kubernetes_cluster_node. Closes #168

### LINKED ISSUE #168: Add table alicloud_cs_kubernetes_cluster_node
**References**
https://api.alibabacloud.com/?spm=5176.14113079.top-nav.19.c4fa7815F269Iq#/?product=CS&version=2015-12-15&api=DescribeClusterNodes&tab=DEMO&lang=JAVA

### PR DESCRIPTION
**Pull request by rajeshbal65**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
No env file present for the current environment:  staging
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/alicloud_cs_kubernetes_cluster_node []

PRETEST: tests/alicloud_cs_kubernetes_cluster_node

TEST: tests/alicloud_cs_kubernetes_cluster_node
Running terraform
alicloud_vpc.named_test_resource: Creating...
alicloud_vpc.named_test_resource: Creation complete after 8s [id=vpc-0xiijcnwu74hh7xk5apx1]
alicloud_vswitch.named_test_resource: Creating...
alicloud_vswitch.named_test_resource: Creation complete after 8s [id=vsw-0xic68vjzqhlrxh8nz3f7]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Creating...
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [1m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [1m10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [1m20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [1m30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [1m40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [1m50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [2m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [2m10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [2m20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [2m30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [2m40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [2m50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [3m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [3m10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [3m20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [3m30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [3m40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [3m50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [4m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [4m10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [4m20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [4m30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [4m40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [4m50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [5m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [5m10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [5m20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [5m30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [5m40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [5m50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [6m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [6m10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [6m20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [6m30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [6m40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [6m50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [7m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [7m10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [7m20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [7m30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [7m40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [7m50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [8m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [8m10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [8m20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [8m30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [8m40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [8m50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [9m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [9m10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Creation complete after 9m11s [id=c594fb04af3774bef83ed733cfd7597ff]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

cluster_id = "c594fb04af3774bef83ed733cfd7597ff"
instance_id = "i-0xijd5ee7h54dna7gx6i"
instance_name = "worker-k8s-for-cs-c594fb04af3774bef83ed733cfd7597ff"
ip_address = "10.1.1.55"
region = "us-east-1"
resource_aka = "acs:cs:us-east-1:5982111499156037:container/turbottest99723"
resource_name = "turbottest99723"

Running SQL query: test-get-query.sql
2021-06-15T23:19:37.690+0530 [TRACE] steampipe: db.EnsureDbAndStartService start
2021-06-15T23:19:37.690+0530 [TRACE] steampipe: ensure installed
2021-06-15T23:19:37.691+0530 [TRACE] steampipe: ensured installed
2021-06-15T23:19:37.691+0530 [TRACE] steampipe: GetRunStatus - loadRunningInstanceInfo returned nil
2021-06-15T23:19:37.691+0530 [TRACE] steampipe: start service
2021-06-15T23:19:38.629+0530 [TRACE] steampipe: createDbClient
2021-06-15T23:19:38.629+0530 [TRACE] steampipe: status:  &{37397 9193 [localhost 127.0.0.1] local query 115a-4ba8-b16f steampipe steampipe}
2021-06-15T23:19:38.629+0530 [TRACE] steampipe: Connection string:  host=localhost port=9193 user=steampipe dbname=steampipe sslmode=disable
2021-06-15T23:19:39.130+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/slack@latest
2021-06-15T23:19:39.163+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/alicloud@latest
2021-06-15T23:19:39.303+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/aws@latest
2021-06-15T23:19:39.522+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/azure@latest
2021-06-15T23:19:39.606+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/gcp@latest
2021-06-15T23:19:39.684+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/ipstack@latest
2021-06-15T23:19:39.716+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/oci@latest
2021-06-15T23:19:39.776+0530 [TRACE] steampipe: RefreshConnections, updates: &{Update:map[] Delete:map[] MissingPlugins:[] RequiredConnections:map[alicloud:0xc00008c640 aws:0xc00008c690 azure:0xc00008c6e0 gcp:0xc00008c730 ipstack:0xc00008c780 oci:0xc00008c7d0 slack:0xc00008c320]}
2021-06-15T23:19:39.776+0530 [TRACE] steampipe: no connections to update
2021-06-15T23:19:39.776+0530 [TRACE] steampipe: createDbClient
2021-06-15T23:19:39.776+0530 [TRACE] steampipe: status:  &{37397 9193 [localhost 127.0.0.1] local query 115a-4ba8-b16f steampipe steampipe}
2021-06-15T23:19:39.776+0530 [TRACE] steampipe: Connection string:  host=localhost port=9193 user=root dbname=steampipe sslmode=disable
2021-06-15T23:19:39.790+0530 [TRACE] steampipe: createDbClient
2021-06-15T23:19:39.790+0530 [TRACE] steampipe: status:  &{37397 9193 [localhost 127.0.0.1] local query 115a-4ba8-b16f steampipe steampipe}
2021-06-15T23:19:39.790+0530 [TRACE] steampipe: Connection string:  host=localhost port=9193 user=steampipe dbname=steampipe sslmode=disable
2021-06-15T23:19:41.513+0530 [TRACE] steampipe: shutdown
2021-06-15T23:19:41.513+0530 [TRACE] steampipe: StopDB false
2021-06-15T23:19:41.514+0530 [TRACE] steampipe: signal sent terminated
[
  {
    "cluster_id": "c594fb04af3774bef83ed733cfd7597ff",
    "instance_id": "i-0xijd5ee7h54dna7gx6i",
    "instance_name": "worker-k8s-for-cs-c594fb04af3774bef83ed733cfd7597ff"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
2021-06-15T23:19:41.595+0530 [TRACE] steampipe: db.EnsureDbAndStartService start
2021-06-15T23:19:41.595+0530 [TRACE] steampipe: ensure installed
2021-06-15T23:19:41.596+0530 [TRACE] steampipe: ensured installed
2021-06-15T23:19:41.596+0530 [TRACE] steampipe: GetRunStatus - loadRunningInstanceInfo returned nil
2021-06-15T23:19:41.596+0530 [TRACE] steampipe: start service
2021-06-15T23:19:42.228+0530 [TRACE] steampipe: createDbClient
2021-06-15T23:19:42.228+0530 [TRACE] steampipe: status:  &{37427 9193 [localhost 127.0.0.1] local query 115a-4ba8-b16f steampipe steampipe}
2021-06-15T23:19:42.228+0530 [TRACE] steampipe: Connection string:  host=localhost port=9193 user=steampipe dbname=steampipe sslmode=disable
2021-06-15T23:19:42.735+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/gcp@latest
2021-06-15T23:19:42.801+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/ipstack@latest
2021-06-15T23:19:42.825+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/oci@latest
2021-06-15T23:19:42.878+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/slack@latest
2021-06-15T23:19:42.906+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/alicloud@latest
2021-06-15T23:19:43.023+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/aws@latest
2021-06-15T23:19:43.200+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/azure@latest
2021-06-15T23:19:43.261+0530 [TRACE] steampipe: RefreshConnections, updates: &{Update:map[] Delete:map[] MissingPlugins:[] RequiredConnections:map[alicloud:0xc000396820 aws:0xc000396870 azure:0xc0003968c0 gcp:0xc000396690 ipstack:0xc0003966e0 oci:0xc000396780 slack:0xc0003967d0]}
2021-06-15T23:19:43.261+0530 [TRACE] steampipe: no connections to update
2021-06-15T23:19:43.261+0530 [TRACE] steampipe: createDbClient
2021-06-15T23:19:43.262+0530 [TRACE] steampipe: status:  &{37427 9193 [localhost 127.0.0.1] local query 115a-4ba8-b16f steampipe steampipe}
2021-06-15T23:19:43.262+0530 [TRACE] steampipe: Connection string:  host=localhost port=9193 user=root dbname=steampipe sslmode=disable
2021-06-15T23:19:43.268+0530 [TRACE] steampipe: createDbClient
2021-06-15T23:19:43.268+0530 [TRACE] steampipe: status:  &{37427 9193 [localhost 127.0.0.1] local query 115a-4ba8-b16f steampipe steampipe}
2021-06-15T23:19:43.268+0530 [TRACE] steampipe: Connection string:  host=localhost port=9193 user=steampipe dbname=steampipe sslmode=disable
2021-06-15T23:19:45.066+0530 [TRACE] steampipe: shutdown
2021-06-15T23:19:45.066+0530 [TRACE] steampipe: StopDB false
2021-06-15T23:19:45.067+0530 [TRACE] steampipe: signal sent terminated
[
  {
    "cluster_id": "c594fb04af3774bef83ed733cfd7597ff",
    "instance_id": "i-0xijd5ee7h54dna7gx6i",
    "instance_name": "worker-k8s-for-cs-c594fb04af3774bef83ed733cfd7597ff"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
2021-06-15T23:19:45.147+0530 [TRACE] steampipe: db.EnsureDbAndStartService start
2021-06-15T23:19:45.147+0530 [TRACE] steampipe: ensure installed
2021-06-15T23:19:45.148+0530 [TRACE] steampipe: ensured installed
2021-06-15T23:19:45.148+0530 [TRACE] steampipe: GetRunStatus - loadRunningInstanceInfo returned nil
2021-06-15T23:19:45.148+0530 [TRACE] steampipe: start service
2021-06-15T23:19:45.771+0530 [TRACE] steampipe: createDbClient
2021-06-15T23:19:45.771+0530 [TRACE] steampipe: status:  &{37455 9193 [localhost 127.0.0.1] local query 115a-4ba8-b16f steampipe steampipe}
2021-06-15T23:19:45.771+0530 [TRACE] steampipe: Connection string:  host=localhost port=9193 user=steampipe dbname=steampipe sslmode=disable
2021-06-15T23:19:46.276+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/azure@latest
2021-06-15T23:19:46.337+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/gcp@latest
2021-06-15T23:19:46.405+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/ipstack@latest
2021-06-15T23:19:46.429+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/oci@latest
2021-06-15T23:19:46.480+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/slack@latest
2021-06-15T23:19:46.507+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/alicloud@latest
2021-06-15T23:19:46.622+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/aws@latest
2021-06-15T23:19:46.798+0530 [TRACE] steampipe: RefreshConnections, updates: &{Update:map[] Delete:map[] MissingPlugins:[] RequiredConnections:map[alicloud:0xc0000be730 aws:0xc0000be780 azure:0xc0000be2d0 gcp:0xc0000be5f0 ipstack:0xc0000be640 oci:0xc0000be690 slack:0xc0000be6e0]}
2021-06-15T23:19:46.798+0530 [TRACE] steampipe: no connections to update
2021-06-15T23:19:46.798+0530 [TRACE] steampipe: createDbClient
2021-06-15T23:19:46.799+0530 [TRACE] steampipe: status:  &{37455 9193 [localhost 127.0.0.1] local query 115a-4ba8-b16f steampipe steampipe}
2021-06-15T23:19:46.799+0530 [TRACE] steampipe: Connection string:  host=localhost port=9193 user=root dbname=steampipe sslmode=disable
2021-06-15T23:19:46.805+0530 [TRACE] steampipe: createDbClient
2021-06-15T23:19:46.805+0530 [TRACE] steampipe: status:  &{37455 9193 [localhost 127.0.0.1] local query 115a-4ba8-b16f steampipe steampipe}
2021-06-15T23:19:46.805+0530 [TRACE] steampipe: Connection string:  host=localhost port=9193 user=steampipe dbname=steampipe sslmode=disable
2021-06-15T23:19:48.564+0530 [TRACE] steampipe: shutdown
2021-06-15T23:19:48.564+0530 [TRACE] steampipe: StopDB false
2021-06-15T23:19:48.565+0530 [TRACE] steampipe: signal sent terminated
null
✔ PASSED

POSTTEST: tests/alicloud_cs_kubernetes_cluster_node

TEARDOWN: tests/alicloud_cs_kubernetes_cluster_node

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
### Basic info

```sql
select
  node_name,
  cluster_id,
  state,
  creation_time,
  instance_id
from
  alicloud_cs_kubernetes_cluster_node;

+--------------------------+-----------------------------------+---------+---------------------+------------------------+
| node_name                | cluster_id                        | state   | creation_time       | instance_id            |
+--------------------------+-----------------------------------+---------+---------------------+------------------------+
| cn-beijing.10.1.75.178   | c3bab6a22f20d4220b34490a812c5ca8c | running | 2021-06-18 07:03:30 | i-2zej2a3evssx0ocmbjpq |
| cn-beijing.10.1.75.179   | c3bab6a22f20d4220b34490a812c5ca8c | running | 2021-06-18 07:03:29 | i-2zej2a3evssx0ocmbjpr |
| ap-south-1.172.18.243.71 | c9ef33d4ff0f24a58936734712b6c3b13 | running | 2021-06-17 13:53:37 | i-a2d7bobadx0zxi63y26r |
| ap-south-1.172.18.243.72 | c9ef33d4ff0f24a58936734712b6c3b13 | running | 2021-06-17 13:53:37 | i-a2d7bobadx0zxi63y26s |
| ap-south-1.172.18.243.70 | c9ef33d4ff0f24a58936734712b6c3b13 | running | 2021-06-17 13:53:37 | i-a2d7bobadx0zxi63y26t |
+--------------------------+-----------------------------------+---------+---------------------+------------------------+
```

### List of worker nodes

```sql
select
  node_name,
  instance_id,
  instance_name,
  instance_role
from
  alicloud_cs_kubernetes_cluster_node
where
  instance_role = 'Worker';

+--------------------------+------------------------+-----------------------------------------------------+---------------+
| node_name                | instance_id            | instance_name                                       | instance_role |
+--------------------------+------------------------+-----------------------------------------------------+---------------+
| cn-beijing.10.1.75.179   | i-2zej2a3evssx0ocmbjpr | worker-k8s-for-cs-c3bab6a22f20d4220b34490a812c5ca8c | Worker        |
| cn-beijing.10.1.75.178   | i-2zej2a3evssx0ocmbjpq | worker-k8s-for-cs-c3bab6a22f20d4220b34490a812c5ca8c | Worker        |
| ap-south-1.172.18.243.72 | i-a2d7bobadx0zxi63y26s | worker-k8s-for-cs-c9ef33d4ff0f24a58936734712b6c3b13 | Worker        |
| ap-south-1.172.18.243.70 | i-a2d7bobadx0zxi63y26t | worker-k8s-for-cs-c9ef33d4ff0f24a58936734712b6c3b13 | Worker        |
| ap-south-1.172.18.243.71 | i-a2d7bobadx0zxi63y26r | worker-k8s-for-cs-c9ef33d4ff0f24a58936734712b6c3b13 | Worker        |
+--------------------------+------------------------+-----------------------------------------------------+---------------+
```

### List ecs instance info of nodes

```sql
select
  node_name,
  instance_id,
  instance_name,
  instance_role,
  instance_type,
  instance_charge_type,
  instance_type_family
from
  alicloud_cs_kubernetes_cluster_node;

+--------------------------+------------------------+-----------------------------------------------------+---------------+---------------+----------------------+----------------------+
| node_name                | instance_id            | instance_name                                       | instance_role | instance_type | instance_charge_type | instance_type_family |
+--------------------------+------------------------+-----------------------------------------------------+---------------+---------------+----------------------+----------------------+
| ap-south-1.172.18.243.71 | i-a2d7bobadx0zxi63y26r | worker-k8s-for-cs-c9ef33d4ff0f24a58936734712b6c3b13 | Worker        | ecs.i2.xlarge | PostPaid             | ecs.i2               |
| ap-south-1.172.18.243.70 | i-a2d7bobadx0zxi63y26t | worker-k8s-for-cs-c9ef33d4ff0f24a58936734712b6c3b13 | Worker        | ecs.i2.xlarge | PostPaid             | ecs.i2               |
| ap-south-1.172.18.243.72 | i-a2d7bobadx0zxi63y26s | worker-k8s-for-cs-c9ef33d4ff0f24a58936734712b6c3b13 | Worker        | ecs.i2.xlarge | PostPaid             | ecs.i2               |
| cn-beijing.10.1.75.179   | i-2zej2a3evssx0ocmbjpr | worker-k8s-for-cs-c3bab6a22f20d4220b34490a812c5ca8c | Worker        | ecs.sn2.large | PostPaid             | ecs.sn2              |
| cn-beijing.10.1.75.178   | i-2zej2a3evssx0ocmbjpq | worker-k8s-for-cs-c3bab6a22f20d4220b34490a812c5ca8c | Worker        | ecs.sn2.large | PostPaid             | ecs.sn2              |
+--------------------------+------------------------+-----------------------------------------------------+---------------+---------------+----------------------+----------------------+
```

### Count of node per instance

```sql
select
  instance_id,
  count(*) as node_count
from
  alicloud_cs_kubernetes_cluster_node
group by
  instance_id;

+------------------------+------------+
| instance_id            | node_count |
+------------------------+------------+
| i-2zej2a3evssx0ocmbjpq | 1          |
| i-2zej2a3evssx0ocmbjpr | 1          |
| i-a2d7bobadx0zxi63y26r | 1          |
| i-a2d7bobadx0zxi63y26s | 1          |
| i-a2d7bobadx0zxi63y26t | 1          |
+------------------------+------------+
```
```
</details>

### CODE REVIEW

File: alicloud-test/tests/alicloud_cs_kubernetes_cluster_node/test-get-expected.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[
+	{
```
Comment (bigdatasourav) [inline alicloud-test/tests/alicloud_cs_kubernetes_cluster_node/test-get-expected.json:2]: format is not correct , it should be 2 spaces.

File: alicloud-test/tests/alicloud_cs_kubernetes_cluster_node/test-list-expected.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[
+	{
```
Comment (bigdatasourav) [inline alicloud-test/tests/alicloud_cs_kubernetes_cluster_node/test-list-expected.json:2]: It should be 2 spaces.

File: alicloud-test/tests/alicloud_cs_kubernetes_cluster_node/variables.tf [hunk 3]
```diff
@@ -0,0 +1,78 @@
+
```
Comment (bigdatasourav) [inline alicloud-test/tests/alicloud_cs_kubernetes_cluster_node/variables.tf:1]:
```suggestion
```

File: alicloud/table_alicloud_cs_kubernetes_cluster_node.go [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
+				Name:        "expired_time",
+				Description: "The expiration time of the node.",
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
+				Name:        "node_status",
+				Description: "Indicates whether the node is ready in the ACK cluster. Valid values: true, false",
```
Comment (bigdatasourav) [inline alicloud/table_alicloud_cs_kubernetes_cluster_node.go:59]:
```suggestion
				Description: "Indicates whether the node is ready in the ACK cluster. Valid values: true, false.",
```

File: alicloud/table_alicloud_cs_kubernetes_cluster_node.go [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+func listCsKubernetesClusterNodes(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	region := plugin.GetMatrixItem(ctx)[matrixKeyRegion].(string)
+
+	// Create service connection
+	client, err := ContainerService(ctx, d, region)
+	if err != nil {
+		plugin.Logger(ctx).Error("listCsKubernetesClusterNodes", "connection_error", err)
+		return nil, err
+	}
+	clusterId := h.Item.(map[string]interface{})["cluster_id"].(string)
```
Comment (bigdatasourav) [inline alicloud/table_alicloud_cs_kubernetes_cluster_node.go:177]:
```suggestion

	clusterId := h.Item.(map[string]interface{})["cluster_id"].(string)
```

File: alicloud/table_alicloud_cs_kubernetes_cluster_node.go [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+	request.Scheme = "https"
+	request.ClusterId = clusterId
+
+	response, err := client.DescribeClusterNodes(request)
+	if err != nil {
+		plugin.Logger(ctx).Error("listCsKubernetesClusterNodes", "query_error", err, "request", request)
+		return nil, err
+	}
+	for _, node := range response.Nodes {
+		d.StreamLeafListItem(ctx, node)
```
Comment (bigdatasourav) [inline alicloud/table_alicloud_cs_kubernetes_cluster_node.go:188]:
```suggestion
		d.StreamListItem(ctx, node)
```

File: alicloud/table_alicloud_cs_kubernetes_cluster_node.go [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+	// Create service connection
+	client, err := ContainerService(ctx, d, matrixRegion)
+	if err != nil {
+		plugin.Logger(ctx).Error("getCsKubernetesClusterNode", "connection_error", err)
+		return nil, err
+	}
+
+	clusterId := d.KeyColumnQuals["cluster_id"].GetStringValue()
+	instanceId := d.KeyColumnQuals["instance_id"].GetStringValue()
+
```
Comment (bigdatasourav) [inline alicloud/table_alicloud_cs_kubernetes_cluster_node.go:208]: can we put a check here for clusterId and InstanceId

File: alicloud/table_alicloud_cs_kubernetes_cluster_node.go [hunk 8]
```diff
@@ -0,0 +1,258 @@
...
+			return item, nil
+		}
+	}
+
+	return nil, nil
+}
+
+func getCsKubernetesClusterNodeAka(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getCsKubernetesClusterNodeAka")
+
```
Comment (bigdatasourav) [inline alicloud/table_alicloud_cs_kubernetes_cluster_node.go:230]:
```suggestion
```

File: alicloud/table_alicloud_cs_kubernetes_cluster_node.go [hunk 9]
```diff
@@ -0,0 +1,258 @@
...
+				Hydrate:     getCsKubernetesClusterNodeAka,
+				Transform:   transform.FromValue(),
+			},
+
+			// Alicloud standard columns
+			{
+				Name:        "region",
+				Description: ColumnDescriptionRegion,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.From(getNodeRegion),
```
Comment (bigdatasourav) [inline alicloud/table_alicloud_cs_kubernetes_cluster_node.go:153]:
```suggestion
				Transform:   transform.From(clusterNodeRegion),
```

File: alicloud/table_alicloud_cs_kubernetes_cluster_node.go [hunk 10]
```diff
@@ -0,0 +1,258 @@
...
+			{
+				Name:        "node_name",
+				Description: "The name of the node in the ACK cluster.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "cluster_id",
+				Description: "The ID of the cluster that the node pool belongs to.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.From(getClusterId),
```
Comment (bigdatasourav) [inline alicloud/table_alicloud_cs_kubernetes_cluster_node.go:40]:
```suggestion
				Transform:   transform.From(clusterIdFromInstanceName),
```

File: alicloud-test/tests/alicloud_cs_kubernetes_cluster_node/variables.tf [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+output "resource_name" {
+  value = var.resource_name
+}
+
+output "region" {
+  value = var.alicloud_region
+}
+
+output "resource_aka" {
+  value = "acs:cs:${var.alicloud_region}:${data.alicloud_caller_identity.current.account_id}:container/${var.resource_name}"
```
Comment (bigdatasourav) [inline alicloud-test/tests/alicloud_cs_kubernetes_cluster_node/variables.tf:76]:
```suggestion
  value = "acs:cs:${var.alicloud_region}:${data.alicloud_caller_identity.current.account_id}:node/${var.resource_name}"
```

**Review by bigdatasourav — APPROVED** [review]

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**